### ISSUE TITLE
libamdhip64 not found

### ISSUE THREAD

**Issue by schoonovernumerics** [ROCmSoftwarePlatform/hipfort#13]
Operating system : Ubuntu 18.04
Compiler : GCC, GFortran (7.4.0)
GPU : MI25 Radeon

I've created a simple Fortran application to reproduce an issue with compiling Fortran files with `hipfc`. The program is shown below
```
PROGRAM Test
USE hip
USE iso_c_binding

IMPLICIT NONE
  INTEGER(c_size_t), PARAMETER :: N=100
  TYPE(c_ptr) :: a_dev
      CALL hipCheck(hipMalloc(a_dev, 4*N))
      CALL hipCheck(hipFree(a_dev))
END PROGRAM Test
```
I compile with `hipfc`, which shows the following error
```
hipfc -v test.f90 

/usr/bin/gfortran -cpp -I/opt/rocm/hipfort/include/amdgcn   test.f90 -L/opt/rocm/lib -lamdhip64 -Wl,-rpath=/opt/rocm/lib  -lstdc++ -L/opt/rocm/hipfort/lib -lhipfort-amdgcn -o /home/joe/self-fluids/src/self/minimal/a.out
/usr/bin/ld: cannot find -lamdhip64
collect2: error: ld returned 1 exit status
ERROR:  The following command failed with return code 1.
        /usr/bin/gfortran -cpp -I/opt/rocm/hipfort/include/amdgcn   test.f90 -L/opt/rocm/lib -lamdhip64 -Wl,-rpath=/opt/rocm/lib  -lstdc++ -L/opt/rocm/hipfort/lib -lhipfort-amdgcn -o /home/joe/self-fluids/src/self/minimal/a.out
```
I've confirmed that there are no libraries installed that correspond to libamdhip64.so under /opt/rocm/lib.
If this library is a dependency, we need to build out documentation that lists this as a dependency. Removing this flag in hipfc directly resolves the issue.

**Comment by pbauman** [comment]
@schoonovernumerics What version of ROCm are you on? If you don't know, could you please report the output of `cat /opt/rocm/.info/version`? I suspect you're on a version earlier than 3.5. ROCm 3.5 and later changed, among many other things, the runtime library to `libamdhip64.so`.

**Comment by schoonovernumerics** [comment]
@pbauman you're right. I'm at 3.3 . Do we want to note the minimum requirement of ROCm 3.5 in the README ?
I'll go through  an upgrade and confirm if this resolves the issue.

**Comment by schoonovernumerics** [comment]
When attempting to upgrade to 3.5, I removed the existing install of rocm3.3 and ran
```
sudo apt install rocm-dkms3.5.0
```
And I receive a number of 404 errors, such as
```
E: Failed to fetch http://repo.radeon.com/rocm/apt/debian/pool/main/r/rocm-dkms3.5.0/rocm-dkms3.5.0_3.5.1-34_amd64.deb  404  Not Found
```
Looking around at http://repo.radeon.com/rocm/apt/debian/pool/main/r/rocm-dkms3.5.0/, I can confirm the 3.5.1-34_amd64 file isn't there for rocm-dkms. A similar issue happens with all of the other dependencies.

I ran a dist-upgrade and update, rebooted, and this secondary issue with rocm3.5.0 installation persists.
